steno/src/model/classifier/classifier.cpp

```diff
@@ -5,35 +5,70 @@
-                                      const std::vector<SphericalAngleRect> &imagePositions,
-                                      const float &rangeThreshold)
+std::vector<int> Classifier::getSourcesToKeep(const std::vector<SourcePosition> &audioPositions,
+                                              const std::vector<SphericalAngleRect> &imagePositions,
+                                              const float &rangeThreshold)
 {
-    std::vector<int> sourcesToSuppress;
+    std::vector<int> sourcesToKeep;
+    std::vector<std::pair<int, int>> audioImagePairs =
+        getAudioImagePairs(audioPositions, imagePositions, rangeThreshold);
+
+    for (size_t i = 0; i < audioImagePairs.size(); i++)
+    {
+        std::pair<int, int> pair = audioImagePairs[i];
+
+        // Check if the index to keep is not already in the vector
+        if (std::find(sourcesToKeep.begin(), sourcesToKeep.end(), pair.first) == sourcesToKeep.end())
+        {
+            sourcesToKeep.push_back(pair.first);
+        }
+    }
+
+    return sourcesToKeep;
+}
+
+/**
+ * @brief Compares the spatial positions of audio sources and images to identify audio/image pairs
+ * @param audioPositions - audio positions from odas localization.
+ * @param imagePositions - image positions to compare with audio positions.
+ * @param rangeThreshold - distance threshold
+ * @return vector of pairs of index that represents the audio sources and images that are at the same spatial position
+ */
+std::vector<std::pair<int, int>> Classifier::getAudioImagePairs(const std::vector<SourcePosition> &audioPositions,
+                                                                const std::vector<SphericalAngleRect> &imagePositions,
+                                                                const float &rangeThreshold)
+{
+    std::pair<int, int> audioImagePair;
+    std::vector<std::pair<int, int>> audioImagePairs;
 
     for (size_t i = 0; i < audioPositions.size(); i++)
     {
-        if (audioPositions[i].elevation > 0)
+        for (size_t j = 0; j < imagePositions.size(); j++)
         {
-            for (size_t j = 0; j < imagePositions.size(); j++)
+            // Set the azimuth counter-clockwise to match the audio
+            float imagePositionAzimuth = 2.0 * M_PI - imagePositions[j].azimuth;
+
+            if (std::abs(audioPositions[i].azimuth - imagePositionAzimuth) < rangeThreshold &&
+                std::abs(audioPositions[i].elevation - imagePositions[j].elevation) < rangeThreshold)
             {
-                if (std::abs(audioPositions[i].azimuth - imagePositions[j].azimuth) > rangeThreshold ||
-                    std::abs(audioPositions[i].elevation - imagePositions[j].elevation) > rangeThreshold)
+                audioImagePair = std::make_pair(i, j);
+
+                // Check if the audio/image pair is already in the vector
+                if (std::find(audioImagePairs.begin(), audioImagePairs.end(), audioImagePair) == audioImagePairs.end())
```

Comment:
This check is probably unnecessary, as the vector is empty at the start of the function and a <i,j> pair can only occur once in the loops